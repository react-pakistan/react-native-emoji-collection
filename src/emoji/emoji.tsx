import React, { ReactElement } from 'react';
import { SvgXml } from 'react-native-svg';

export const Emoji = ({
  emoji,
  fill,
} : IEmojiProps) : ReactElement => {
  const iconSvg = require('../xml')[emoji]();

  return (
    <SvgXml
      width="100%"
      height="100%"
      xml={`${iconSvg}`}
      fill={fill}
    />
  );
};

export interface IEmojiProps {
  emoji : string;
  fill? : string;
}
