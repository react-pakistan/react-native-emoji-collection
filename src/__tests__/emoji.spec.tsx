import React from 'react';
import renderer from 'react-test-renderer';
import { Emoji } from '../emoji';

describe('Testing Emoji component', () => {
  it('matches Emoji snapshot,', () => {
    const wrapper = renderer.create(
      <Emoji emoji="BlockchainColor1" />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
