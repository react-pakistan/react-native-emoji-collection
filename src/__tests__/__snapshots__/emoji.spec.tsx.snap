// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Emoji component matches Emoji snapshot, 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="100%"
  bbWidth="100%"
  focusable={false}
  height="100%"
  id="emoji"
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
  vbHeight={72}
  vbWidth={72}
  width="100%"
  xml="<svg id=\\"emoji\\" viewBox=\\"0 0 72 72\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <g id=\\"color\\">
    <path fill=\\"#A57939\\" stroke=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\" d=\\"M59,36c-0.25-0.75-0.71-2.1-2-3c-1.56-1.08-3.63-1.01-4-1c-0.76,0.03-1.18,0.16-2,0c-0.58-0.12-1.53-0.3-2-1 c-0.4-0.59-0.15-1.08,0-3c0.12-1.51,0.17-2.27,0-3c-0.37-1.58-1.49-2.56-2-3c-1.05-0.92-2.38-1.56-5-2c-1.82-0.31-4.75-0.6-9,0 c-2.15,0.3-5.46,0.87-8,1.72c-1.77,0.58-3.74,1.41-6,3c-0.02,0.01-0.04,0.02-0.05,0.03c-3.44,2.24-5.39,6.2-5.22,10.31 C14.64,57.13,54.56,59.91,59,41C59.09,40.4,59.79,38.35,59,36z M38.21,30.12c-1.53,0-2.76-1.24-2.76-2.76 c0-1.53,1.23-2.76,2.76-2.76h2.71c1.52,0,2.76,1.23,2.76,2.76c0,0.76-0.31,1.45-0.81,1.95s-1.19,0.81-1.95,0.81H38.21z\\"/>
    <circle cx=\\"20\\" cy=\\"33\\" r=\\"3\\" fill=\\"#61B2E4\\" stroke=\\"none\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
    <circle cx=\\"25\\" cy=\\"42\\" r=\\"3\\" fill=\\"#5C9E31\\" stroke=\\"none\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
    <circle cx=\\"35\\" cy=\\"45\\" r=\\"3\\" fill=\\"#FCEA2B\\" stroke=\\"none\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
    <circle cx=\\"45\\" cy=\\"44\\" r=\\"3\\" fill=\\"#D22F27\\" stroke=\\"none\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
  </g>
  <g id=\\"hair\\"/>
  <g id=\\"skin\\"/>
  <g id=\\"skin-shadow\\"/>
  <g id=\\"line\\">
    <path fill=\\"none\\" stroke=\\"#000000\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\" d=\\"M59,36c-0.25-0.75-0.71-2.1-2-3c-1.56-1.08-3.63-1.01-4-1c-0.76,0.03-1.18,0.16-2,0c-0.58-0.12-1.53-0.3-2-1 c-0.4-0.59-0.15-1.08,0-3c0.12-1.51,0.17-2.27,0-3c-0.37-1.58-1.49-2.56-2-3c-1.05-0.92-2.38-1.56-5-2c-1.82-0.31-4.75-0.6-9,0 c-2.15,0.3-5.46,0.87-8,1.72c-1.77,0.58-3.74,1.41-6,3c-0.02,0.01-0.04,0.02-0.05,0.03c-3.44,2.24-5.39,6.2-5.22,10.31 C14.64,57.13,54.56,59.91,59,41C59.09,40.4,59.79,38.35,59,36z M38.21,30.12c-1.53,0-2.76-1.24-2.76-2.76 c0-1.53,1.23-2.76,2.76-2.76h2.71c1.52,0,2.76,1.23,2.76,2.76c0,0.76-0.31,1.45-0.81,1.95s-1.19,0.81-1.95,0.81H38.21z\\"/>
    <circle cx=\\"20\\" cy=\\"33\\" r=\\"3\\" fill=\\"none\\" stroke=\\"#000000\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
    <circle cx=\\"25\\" cy=\\"42\\" r=\\"3\\" fill=\\"none\\" stroke=\\"#000000\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
    <circle cx=\\"35\\" cy=\\"45\\" r=\\"3\\" fill=\\"none\\" stroke=\\"#000000\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
    <circle cx=\\"45\\" cy=\\"44\\" r=\\"3\\" fill=\\"none\\" stroke=\\"#000000\\" stroke-miterlimit=\\"10\\" stroke-width=\\"2\\"/>
  </g>
</svg>
"
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup>
    <RNSVGGroup
      name="color"
    >
      <RNSVGPath
        d="M59,36c-0.25-0.75-0.71-2.1-2-3c-1.56-1.08-3.63-1.01-4-1c-0.76,0.03-1.18,0.16-2,0c-0.58-0.12-1.53-0.3-2-1 c-0.4-0.59-0.15-1.08,0-3c0.12-1.51,0.17-2.27,0-3c-0.37-1.58-1.49-2.56-2-3c-1.05-0.92-2.38-1.56-5-2c-1.82-0.31-4.75-0.6-9,0 c-2.15,0.3-5.46,0.87-8,1.72c-1.77,0.58-3.74,1.41-6,3c-0.02,0.01-0.04,0.02-0.05,0.03c-3.44,2.24-5.39,6.2-5.22,10.31 C14.64,57.13,54.56,59.91,59,41C59.09,40.4,59.79,38.35,59,36z M38.21,30.12c-1.53,0-2.76-1.24-2.76-2.76 c0-1.53,1.23-2.76,2.76-2.76h2.71c1.52,0,2.76,1.23,2.76,2.76c0,0.76-0.31,1.45-0.81,1.95s-1.19,0.81-1.95,0.81H38.21z"
        fill={4289034553}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
            "strokeMiterlimit",
          ]
        }
        stroke={null}
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
      <RNSVGCircle
        cx={20}
        cy={33}
        fill={4284592868}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeMiterlimit",
          ]
        }
        r={3}
        stroke={null}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
      <RNSVGCircle
        cx={25}
        cy={42}
        fill={4284259889}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
            "strokeMiterlimit",
          ]
        }
        r={3}
        stroke={null}
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
      <RNSVGCircle
        cx={35}
        cy={45}
        fill={4294765099}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeMiterlimit",
          ]
        }
        r={3}
        stroke={null}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
      <RNSVGCircle
        cx={45}
        cy={44}
        fill={4291964711}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeMiterlimit",
          ]
        }
        r={3}
        stroke={null}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
    </RNSVGGroup>
    <RNSVGGroup
      name="hair"
    />
    <RNSVGGroup
      name="skin"
    />
    <RNSVGGroup
      name="skin-shadow"
    />
    <RNSVGGroup
      name="line"
    >
      <RNSVGPath
        d="M59,36c-0.25-0.75-0.71-2.1-2-3c-1.56-1.08-3.63-1.01-4-1c-0.76,0.03-1.18,0.16-2,0c-0.58-0.12-1.53-0.3-2-1 c-0.4-0.59-0.15-1.08,0-3c0.12-1.51,0.17-2.27,0-3c-0.37-1.58-1.49-2.56-2-3c-1.05-0.92-2.38-1.56-5-2c-1.82-0.31-4.75-0.6-9,0 c-2.15,0.3-5.46,0.87-8,1.72c-1.77,0.58-3.74,1.41-6,3c-0.02,0.01-0.04,0.02-0.05,0.03c-3.44,2.24-5.39,6.2-5.22,10.31 C14.64,57.13,54.56,59.91,59,41C59.09,40.4,59.79,38.35,59,36z M38.21,30.12c-1.53,0-2.76-1.24-2.76-2.76 c0-1.53,1.23-2.76,2.76-2.76h2.71c1.52,0,2.76,1.23,2.76,2.76c0,0.76-0.31,1.45-0.81,1.95s-1.19,0.81-1.95,0.81H38.21z"
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
            "strokeMiterlimit",
          ]
        }
        stroke={4278190080}
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
      <RNSVGCircle
        cx={20}
        cy={33}
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeMiterlimit",
          ]
        }
        r={3}
        stroke={4278190080}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
      <RNSVGCircle
        cx={25}
        cy={42}
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
            "strokeMiterlimit",
          ]
        }
        r={3}
        stroke={4278190080}
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
      <RNSVGCircle
        cx={35}
        cy={45}
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeMiterlimit",
          ]
        }
        r={3}
        stroke={4278190080}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
      <RNSVGCircle
        cx={45}
        cy={44}
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeMiterlimit",
          ]
        }
        r={3}
        stroke={4278190080}
        strokeMiterlimit={10}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGGroup>
</RNSVGSvgView>
`;
